misk/src/main/kotlin/misk/flags/DynamicFlagModule.kt

```diff
@@ -0,0 +1,163 @@
+import kotlin.reflect.KFunction
+
+/**
+ * Installs support for dynamic flag. Applications should inherit from this module and
+ * override [configureFlags], calling the various bindXXXFlags methods to make flag
+ * variables available
+ */
+abstract class DynamicFlagModule : KAbstractModule() {
+    override fun configure() {
+        requireBinding(DynamicFlagStore::class.java)
+        configureFlags()
+    }
+
+    protected abstract fun configureFlags()
+
+    inline fun <reified T : Flags> bindFlags(prefix: String = "", qualifier: Annotation? = null) =
+            bindFlags(T::class, prefix, qualifier)
+
+    fun <T : Flags> bindFlags(
+            kclass: KClass<T>,
+            prefix: String = "",
+            qualifier: Annotation? = null
+    ) {
+        val actualQualifier = qualifier ?: if (!prefix.isEmpty()) Names.named(prefix) else null
+        val constructor = kclass.constructors.firstOrNull {
+            it.parameters.size == 1 && it.parameters[0].type.classifier == Flags.Context::class
+        } ?: throw IllegalArgumentException(
+                "$kclass has no single argument constructor taking a Flags.Context")
+
+
+        if (actualQualifier == null) {
+            bind(kclass.java)
+                    .toProvider(FlagsProvider(constructor, prefix))
+                    .asEagerSingleton()
+        } else {
+            bind(kclass.java)
+                    .annotatedWith(actualQualifier)
+                    .toProvider(FlagsProvider(constructor, prefix))
+                    .asEagerSingleton()
+        }
+    }
+
+    fun bindBooleanFlag(name: String, desc: String, qualifier: Annotation = Names.named(name)) {
+        bind<BooleanFlag>()
+                .annotatedWith(qualifier)
+                .toProvider(BooleanFlagProvider(name, desc))
+                .asEagerSingleton()
+    }
+
+    fun bindStringFlag(name: String, desc: String, qualifier: Annotation = Names.named(name)) {
+        bind<StringFlag>()
+                .annotatedWith(qualifier)
+                .toProvider(StringFlagProvider(name, desc))
+                .asEagerSingleton()
+    }
+
+    fun bindIntFlag(name: String, desc: String, qualifier: Annotation = Names.named(name)) {
+        bind<IntFlag>()
+                .annotatedWith(qualifier)
+                .toProvider(IntFlagProvider(name, desc))
+                .asEagerSingleton()
+    }
+
+    fun bindDoubleFlag(name: String, desc: String, qualifier: Annotation = Names.named(name)) {
+        bind<DoubleFlag>()
+                .annotatedWith(qualifier)
+                .toProvider(DoubleFlagProvider(name, desc))
+                .asEagerSingleton()
+    }
+
+    inline fun <reified T : Any> bindJsonFlag(
+            name: String,
+            desc: String,
+            qualifier: Annotation = Names.named(name)
+    ) = bindJsonFlag(T::class, name, desc, qualifier)
+
+    fun <T : Any> bindJsonFlag(
+            kclass: KClass<T>,
+            name: String,
+            desc: String,
```

Comment:
nit: I like the greppability of avoiding abbreviations (description vs. desc)

Reply:
Will fix